pkg/utils.go

```diff
@@ -0,0 +1,47 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+)
+
+func Read(ctx context.Context, fs FS, path string, options *ReaderOptions) (data []byte, err error) {
+	var file *File
+	if file, err = fs.Open(ctx, path, options); err != nil {
+		return nil, fmt.Errorf("unable to open %s: %w", path, err)
+	}
+
+	if data, err = ioutil.ReadAll(file); err != nil {
+		return nil, fmt.Errorf("unable to read %s: %w", path, err)
+	}
+
+	if err = file.Close(); err != nil {
```

Comment:
Should we use `defer` for closing the file after opening it ?

Reply:
Maybe not `defer`, because I want the error to bubble up, but you're right that I should add it before the return of unable to read.

Reply:
Interesting post on the subject: https://www.joeshaw.org/dont-defer-close-on-writable-files/

Reply:
I also usually lean toward `defer`. But for this simple function, this is cleaner/simpler.